src/games/strategy/triplea/ui/screen/UnitsDrawer.java

```diff
@@ -90,6 +91,11 @@ public void draw(final Rectangle bounds, final GameData data, final Graphics2D g
     final Optional<Image> img =
         uiContext.getUnitImageFactory().getImage(type, owner, data, damaged > 0 || bombingUnitDamage > 0, disabled);
 
+    if (!img.isPresent()) {
+      ClientLogger
+          .logError("MISSING IMAGE (this unit or image will be invisible): " + type);
+    }
```

Comment:
So here we log the missing image if we try to draw the missing image on the map.
Missing images can still be found by looking at the 'add units' screen in edit mode, or a purchase screen. Trying to place such a unit then yields the error message (and the game will continue running)